code/modules/antagonists/nightmare/nightmare_organs.dm

```diff
@@ -102,5 +102,30 @@
 		return TRUE
 	return ..()
 
+/obj/item/organ/internal/eyes/night_vision/nightmare
+	name = "piercing red eyes"
+	desc = "Even without their nightmarish owner, looking at these eyes gives you a sense of dread."
+	icon = 'icons/obj/medical/organs/shadow_organs.dmi'
+	///Our associated terrorize spell, for antagonist nightmares
+	var/datum/action/cooldown/spell/pointed/terrorize/terrorize_spell
+
+/obj/item/organ/internal/eyes/night_vision/nightmare/Initialize(mapload)
+	. = ..()
+
+	terrorize_spell = new(src)
+
+/obj/item/organ/internal/eyes/night_vision/nightmare/Insert(mob/living/carbon/reciever, special = FALSE, drop_if_replaced = TRUE)
+	. = ..()
+
+	if(!.)
+		return
+
+	if(reciever.mind?.has_antag_datum(/datum/antagonist/nightmare)) //Only a TRUE NIGHTMARE is worthy of using this ability
+		terrorize_spell.Grant(reciever)
+
+/obj/item/organ/internal/eyes/night_vision/nightmare/Remove(mob/living/carbon/eye_owner, special)
+	QDEL_NULL(terrorize_spell)
+	return ..()
```

Comment:
no, only delete the spell if the eyes are deleted (in Destroy), this will runtime if the eyes are removed and inserted into someone